File: app/src/main/java/com/example/loginsignupapi/LoginActivity.java
package com.example.loginsignupapi;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.example.loginsignupapi.Models.ModelUserLogin;
import com.example.loginsignupapi.ViewModels.UserLoginViewModel;

public class LoginActivity extends AppCompatActivity
{

    String emailPhone,password;
    EditText et_email_login, et_password_login;
    Button btn_log_in;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        et_email_login = findViewById(R.id.et_email_login);
        et_password_login = findViewById(R.id.et_password_login);
        btn_log_in = findViewById(R.id.btn_log_in);

        emailPhone  = et_email_login.getText().toString();
        password = et_password_login.getText().toString();

        btn_log_in.setOnClickListener(view->{
            UserLoginViewModel model = ViewModelProviders.of(this).get(UserLoginViewModel.class);

            model.getLoginResult(emailPhone,password,"1","")
                    .observe(this, new Observer<ModelUserLogin>() {
                        @Override
                        public void onChanged(ModelUserLogin modelUserLogin)
                        {
                            Toast.makeText(LoginActivity.this, modelUserLogin.getMessage() , Toast.LENGTH_SHORT).show();
                        }
                    });
        });

    }
}

File: app/src/main/java/com/example/loginsignupapi/LoginActivity.kt
package com.example.loginsignupapi

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelProviders
import com.example.loginsignupapi.Models.ModelUserLogin
import com.example.loginsignupapi.ViewModels.UserLoginViewModel

class LoginActivity : AppCompatActivity() {
    var emailPhone: String? = null
    var password: String? = null
    var et_email_login: EditText? = null
    var et_password_login: EditText? = null
    var btn_log_in: Button? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        et_email_login = findViewById(R.id.et_email_login)
        et_password_login = findViewById(R.id.et_password_login)
        btn_log_in = findViewById(R.id.btn_log_in)
        emailPhone = et_email_login?.getText().toString()
        password = et_password_login?.getText().toString()
        btn_log_in?.setOnClickListener(View.OnClickListener { view: View? ->
            val model = ViewModelProvider(this).get(UserLoginViewModel::class.java)
            model.getLoginResult(emailPhone, password, "1", "")
                    .observe(this, Observer<ModelUserLogin?>
                    {
                        modelUserLogin -> Toast.makeText(this@LoginActivity, modelUserLogin?.message, Toast.LENGTH_SHORT).show()
                    })
        })
    }
}